cmake/compiler/gcc.cmake

```diff
@@ -50,56 +58,76 @@ foreach(file_name include include-fixed)
   list(APPEND NOSTDINC ${_OUTPUT})
 endforeach()
 
-include(${ZEPHYR_BASE}/cmake/gcc-m-cpu.cmake)
+if("${ZEPHYR_TOOLCHAIN_VARIANT}" STREQUAL "zephyr")
+  include(${ZEPHYR_BASE}/cmake/gcc-m-cpu.cmake)
 
-if("${ARCH}" STREQUAL "arm")
-  list(APPEND TOOLCHAIN_C_FLAGS
-    -mthumb
-    -mcpu=${GCC_M_CPU}
-    )
+  if("${ARCH}" STREQUAL "arm")
+    list(APPEND TOOLCHAIN_C_FLAGS
+      -mthumb
+      -mcpu=${GCC_M_CPU}
+      )
 
-  include(${ZEPHYR_BASE}/cmake/fpu-for-gcc-m-cpu.cmake)
+    include(${ZEPHYR_BASE}/cmake/fpu-for-gcc-m-cpu.cmake)
 
-  if(CONFIG_FLOAT)
-    list(APPEND TOOLCHAIN_C_FLAGS -mfpu=${FPU_FOR_${GCC_M_CPU}})
-    if    (CONFIG_FP_SOFTABI)
-      list(APPEND TOOLCHAIN_C_FLAGS -mfloat-abi=soft)
-    elseif(CONFIG_FP_HARDABI)
-      list(APPEND TOOLCHAIN_C_FLAGS -mfloat-abi=hard)
+    if(CONFIG_FLOAT)
+      list(APPEND TOOLCHAIN_C_FLAGS -mfpu=${FPU_FOR_${GCC_M_CPU}})
+      if    (CONFIG_FP_SOFTABI)
+        list(APPEND TOOLCHAIN_C_FLAGS -mfloat-abi=soft)
+      elseif(CONFIG_FP_HARDABI)
+        list(APPEND TOOLCHAIN_C_FLAGS -mfloat-abi=hard)
+      endif()
     endif()
+  elseif("${ARCH}" STREQUAL "arc")
+    list(APPEND TOOLCHAIN_C_FLAGS
+      -mcpu=${GCC_M_CPU}
+      )
   endif()
-elseif("${ARCH}" STREQUAL "arc")
-  list(APPEND TOOLCHAIN_C_FLAGS
-	  -mcpu=${GCC_M_CPU}
+
+  execute_process(
+    COMMAND ${CMAKE_C_COMPILER} ${TOOLCHAIN_C_FLAGS} --print-libgcc-file-name
+    OUTPUT_VARIABLE LIBGCC_FILE_NAME
+    OUTPUT_STRIP_TRAILING_WHITESPACE
     )
-endif()
 
-execute_process(
-  COMMAND ${CMAKE_C_COMPILER} ${TOOLCHAIN_C_FLAGS} --print-libgcc-file-name
-  OUTPUT_VARIABLE LIBGCC_FILE_NAME
-  OUTPUT_STRIP_TRAILING_WHITESPACE
-  )
+  assert_exists(LIBGCC_FILE_NAME)
+
+  get_filename_component(LIBGCC_DIR ${LIBGCC_FILE_NAME} DIRECTORY)
 
-execute_process(
-  COMMAND ${CMAKE_C_COMPILER} ${TOOLCHAIN_C_FLAGS} --print-multi-directory
-  OUTPUT_VARIABLE NEWLIB_DIR
-  OUTPUT_STRIP_TRAILING_WHITESPACE
-  )
+  assert_exists(LIBGCC_DIR)
 
-assert_exists(LIBGCC_FILE_NAME)
+  LIST(APPEND LIB_INCLUDE_DIR "-L\"${LIBGCC_DIR}\"")
+  LIST(APPEND TOOLCHAIN_LIBS gcc)
 
-get_filename_component(LIBGCC_DIR ${LIBGCC_FILE_NAME} DIRECTORY)
+  if(SYSROOT_DIR)
+    # The toolchain has specified a sysroot dir that we can use to set
+    # the libc path's
+    execute_process(
+      COMMAND ${CMAKE_C_COMPILER} ${TOOLCHAIN_C_FLAGS} --print-multi-directory
+      OUTPUT_VARIABLE NEWLIB_DIR
+      OUTPUT_STRIP_TRAILING_WHITESPACE
+      )
 
-assert_exists(LIBGCC_DIR)
+    set(LIBC_LIBRARY_DIR "\"${SYSROOT_DIR}\"/lib/${NEWLIB_DIR}")
+    set(LIBC_INCLUDE_DIR ${SYSROOT_DIR}/include)
+  endif()
+
+elseif("${ZEPHYR_TOOLCHAIN_VARIANT}" STREQUAL "xcc")
+  # Include XCC standard libraries so that users used to Xplorer IDE can port
+  # their code easily
+  set(TOOLCHAIN_INCLUDES
```

Comment:
Is it necessary for xcc to diverge wrt. discovering libc?

Specifically, does xcc support '--print-libgcc-file-name' ?

If so, it should be used to discover the LIB_INCLUDE_DIR where the 'gcc' toolchain lib is located.

Reply:
xcc supports --print-libgcc-file-name but it does not need to look up libgcc.a. I can see that i am able to compile and link without that. So, have not included that for xcc.